### PR TITLE
repo: Use os-release(5) to detect supported Linux distributions

### PR DESCRIPTION
**Pull request by Conan-Kudo**
All supported Linux distributions provide an `/etc/os-release` file that contains the necessary information for determining supported repository modules to use.

Thus, we can rely on it instead of the deprecated `linux_distribution` method from the `platform` module that will be removed in Python 3.7.

This will also make it much easier to extend to support RPM-based Linux distributions.

Signed-off-by: Neal Gompa <ngompa13@gmail.com>

-----

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
@kyrofa @sergiusens @elopio So, I know what's wrong here, but I don't really want to duplicate myself over half a dozen other places. How should I make this used across the board?

**Comment by kalikiana** [comment]
@Conan-Kudo How about introducing `_get_distro` and moving those into `snapcraft/internal/common.py`?

**Comment by Conan-Kudo** [comment]
@kalikiana I left the name as `_get_os_release_info()` as it's technically not a "distro" specific thing and it's clear on what it's called, but I did move it into common.

Let's see how this goes...

**Comment by Conan-Kudo** [comment]
@kalikiana We're all gravy now!

File: snapcraft/internal/common.py [hunk 1]
```diff
@@ -263,3 +263,12 @@ def get_pkg_config_paths(root, arch_triplet):
     ]
 
     return [p for p in paths if os.path.exists(p)]
+
+
+def _get_os_release_info():
```
Comment (elopio) [inline snapcraft/internal/common.py]: We use the underscore in front of the name to mark something as internal to the file.
This function is called from outside of this file, so please remove the underscore.

**Comment by Conan-Kudo** [comment]
@elopio Done.

**Pull request merged**